### PR TITLE
Fix: add ability to create Partitioned Cookies

### PR DESCRIPTION
**Pull request by nicob-29**
Related to https://github.com/fastify/session/issues/213

- [x] run `npm run test` and `npm run benchmark`
npm run benchmark -> failed
TypeError: redisStoreFactory is not a function
    at createServer (C:\GitHub\fastify-session\benchmark\bench.js:19:24)
    at testFunction (C:\GitHub\fastify-session\benchmark\bench.js:56:18)
    at main (C:\GitHub\fastify-session\benchmark\bench.js:83:15)

- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@fastify/session",
-  "version": "10.6.1",
+  "version": "10.6.2",
```
Comment (Eomm) [inline package.json:3]: Don't bump the version
We will take care of it

File: lib/cookie.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ module.exports = class Cookie {
     this.sameSite = cookie.sameSite || null
     this.domain = cookie.domain || null
     this.httpOnly = cookie.httpOnly !== undefined ? cookie.httpOnly : true
+    this.partitioned = cookie.partitioned ?? null
```
Comment (Eomm) [inline lib/cookie.js:13]: it seems a better default as the prop is trimmed away when serialized

```suggestion
    this.partitioned = cookie.partitioned ?? undefined
```

**Review by Eomm — COMMENTED** [review]
COuld you add a test?

File: README.md [hunk 3]
```diff
@@ -74,6 +74,7 @@ The options object is used to generate the `Set-Cookie` header of the session co
 * `expires` - The expiration `date` used for the `Expires` attribute. If both `expires` and `maxAge` are set, then `maxAge` is used.
 * `sameSite`- The `boolean` or `string` of the `SameSite` attribute. Using `Secure` mode with `auto` attribute will change the behavior of the `SameSite` attribute in `http` mode. The `SameSite` attribute will automatically be set to `Lax` with an `http` request. See this [link](https://www.chromium.org/updates/same-site).
 * `domain` - The `Domain` attribute.
+* `partitioned`- The `boolean` value of the `Partitioned` attribute. Using the Partitioned attribute as part of Cookies Having Independent Partitioned State (CHIPS) to allow cross-site access with a separate cookie used per site.Defaults to false.
```
[the remark this answers is not in the document]
Comment (nicob-29) [reply README.md:77]: it is already here: https://github.com/fastify/fastify-cookie

File: lib/cookie.js [hunk 4]
```diff
@@ -10,6 +10,7 @@ module.exports = class Cookie {
     this.sameSite = cookie.sameSite || null
     this.domain = cookie.domain || null
     this.httpOnly = cookie.httpOnly !== undefined ? cookie.httpOnly : true
+    this.partitioned = cookie.partitioned ?? undefined
```
Comment (gurgunday) [inline lib/cookie.js:13]: nit: this means this for all practical purposes:

```suggestion
    this.partitioned = cookie.partitioned
```

**Comment by nicob-29** [comment]
@Eomm , Tests added

File: test/cookie.test.js [hunk 5]
```diff
@@ -535,6 +562,7 @@ test('Cookie', t => {
     t.equal('path' in json, true)
     t.equal('httpOnly' in json, true)
     t.equal('domain' in json, true)
+    t.equal('partitioned' in cookie, true)
```
Comment (Uzlopak) [inline test/cookie.test.js:565]: Shouldnt it be:

```suggestion
    t.equal('partitioned' in json, true)
```

**Review by Fdawgs — CHANGES_REQUESTED** [review]
The PR title needs to be rewritten as it's not great in its current state. Won't make much sense to users reading release notes.

**Comment by gurgunday** [comment]
What is wrong with this CI?

Rerunning

**Review by gurgunday — APPROVED** [review]

**Comment by gurgunday** [comment]
It's not enabled by default?

Also, I wonder why we're not using fastify/cookie here

**Comment by gurgunday** [comment]
And we might want to label it as experimental or non-standard

File: README.md (hunk 3 above)
Comment (gurgunday) [inline README.md:77]:
```suggestion
* `partitioned` (**experimental**) - The `boolean` value of the `Partitioned` attribute. Using the Partitioned attribute as part of Cookies Having Independent Partitioned State (CHIPS) to allow cross-site access with a separate cookie used per site. Defaults to false.
```

**Comment by jsumners** [comment]
I am completely against making a draft proposal the default implementation of anything.

**Comment by gurgunday** [comment]
Yeah me too, the title is wrong, the PR just adds `partitioned`, it doesn't make it the default

**Review by Fdawgs — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by nicob-29** [comment]
Thank you everyone !

**Pull request merged**